pkg/controller/efs/filesystem/setup.go

```diff
@@ -56,12 +59,29 @@ func SetupFileSystem(mgr ctrl.Manager, o controller.Options) error {
 			managed.WithConnectionPublishers(cps...)))
 }
 
-func isUpToDate(cr *svcapitypes.FileSystem, obj *svcsdk.DescribeFileSystemsOutput) (bool, error) {
+type hooks struct {
+	client efsiface.EFSAPI
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.FileSystem, obj *svcsdk.DescribeFileSystemsOutput) (bool, error) {
 	for _, res := range obj.FileSystems {
 		if awsclients.Int64Value(cr.Spec.ForProvider.ProvisionedThroughputInMibps) != int64(aws.Float64Value(res.ProvisionedThroughputInMibps)) {
 			return false, nil
 		}
 	}
+
+	tagsUpToDate, err := svcutils.AreTagsUpToDate(e.client, cr.Spec.ForProvider.Tags, cr.Status.AtProvider.FileSystemID)
+	if err != nil {
+		return false, err
+	}
+	// update tags here, bc AWS only accepts the regular update-request if there is an actual change included
+	if !tagsUpToDate {
+		err := svcutils.UpdateTagsForResource(e.client, cr.Spec.ForProvider.Tags, cr.Status.AtProvider.FileSystemID)
+		if err != nil {
+			return false, err
+		}
+	}
+
```

Comment:
`Observe` and `isUpToDate` should have no side effects. Tag updates should be in `Update`.

Reply:
As discussed, I now put the tags-update-call to postUpdate(), where we ignore the "no change"-response from AWS to the regular resource-update-call.